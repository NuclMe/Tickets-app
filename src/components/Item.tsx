import React from 'react';
import Box from '@mui/material/Box';
import { TableWrapper } from './TableWrapper';
import { numberWithSpaces } from '../utils';
import Typography from '@mui/material/Typography';

export const Item = (props) => {
  const { price, carrier, segments } = props;

  return (
    <Box
      key={price}
      sx={{
        backgroundColor: 'white',
        padding: '20px',
        borderRadius: '5px',
        boxShadow: '0 2px 4px rgba(0, 0, 0, 0.8)',
      }}
    >
      <Box
        sx={{
          display: 'flex',
          justifyContent: 'space-between',
          alignItems: 'center',
        }}
      >
        <Typography
          sx={{
            fontWeight: 'bold',
            color: '#1976d2',
            fontSize: '24px',
            '@media (max-width: 768px)': {
              fontSize: '18px',
            },
          }}
        >
          {numberWithSpaces(price)} ла
        </Typography>
        <img
          src={`https://pics.avs.io/99/36/${carrier}.png`}
          alt="flightCompanyLogo"
        />
      </Box>
      {segments.map((segment) => (
        <TableWrapper key={segment.duration} segments={segment} />
      ))}
    </Box>
  );
};
