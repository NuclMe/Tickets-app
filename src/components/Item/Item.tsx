import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import type { FC } from 'react';

import { device, fontSizes, colors } from '../../styledVars';
import type { ITicket } from '../../types';
import { numberWithSpaces } from '../../utils';
import { TableWrapper } from '../TableWrapper/TableWrapper';

export const Item: FC<ITicket> = ({ price, carrier, segments }) => (
  <Box
    key={price}
    sx={{
      backgroundColor: 'white',
      padding: '20px',
      borderRadius: '5px',
      boxShadow: '0 2px 4px rgba(0, 0, 0, 0.8)',
    }}
  >
    <Box
      sx={{
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
      }}
    >
      <Typography
        sx={{
          fontWeight: 'bold',
          color: colors.primary,
          fontSize: fontSizes.large,
          [`${device.tablet}`]: {
            fontSize: fontSizes.medium,
          },
        }}
      >
        {numberWithSpaces(price)} ла
      </Typography>
      <img
        src={`https://pics.avs.io/99/36/${carrier}.png`}
        alt="flightCompanyLogo"
      />
    </Box>
    {segments.map((segment) => (
      <TableWrapper key={segment.duration} segments={segment} />
    ))}
  </Box>
);
