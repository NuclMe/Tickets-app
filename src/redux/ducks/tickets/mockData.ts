import type { ITicket } from '../../../types';

export const mockTickets: ITicket[] = [
  {
    price: 35230,
    carrier: 'SU',
    segments: [
      {
        origin: 'MOW',
        destination: 'HKT',
        date: '2018-05-18T02:00:00.000Z',
        stops: [],
        duration: 400,
      },
      {
        origin: 'HKT',
        destination: 'MOW',
        date: '2018-05-18T08:30:00.000Z',
        stops: [],
        duration: 300,
      },
    ],
  },
  {
    price: 30000,
    carrier: 'S7',
    segments: [
      {
        origin: 'MOW',
        destination: 'HKT',
        date: '2018-05-18T02:00:00.000Z',
        stops: ['LED'],
        duration: 600,
      },
      {
        origin: 'HKT',
        destination: 'MOW',
        date: '2018-05-18T08:30:00.000Z',
        stops: ['LED'],
        duration: 200,
      },
    ],
  },
  {
    price: 25000,
    carrier: 'TK',
    segments: [
      {
        origin: 'MOW',
        destination: 'HKT',
        date: '2018-05-18T02:00:00.000Z',
        stops: ['IST', 'DXB'],
        duration: 750,
      },
      {
        origin: 'HKT',
        destination: 'MOW',
        date: '2018-05-18T08:30:00.000Z',
        stops: ['IST', 'DXB'],
        duration: 650,
      },
    ],
  },
  {
    price: 40000,
    carrier: 'BA',
    segments: [
      {
        origin: 'MOW',
        destination: 'HKT',
        date: '2018-05-18T02:00:00.000Z',
        stops: ['LED'],
        duration: 500,
      },
      {
        origin: 'HKT',
        destination: 'MOW',
        date: '2018-05-18T08:30:00.000Z',
        stops: ['LED', 'AMS'],
        duration: 450,
      },
    ],
  },
  {
    price: 45000,
    carrier: 'LH',
    segments: [
      {
        origin: 'MOW',
        destination: 'HKT',
        date: '2018-05-18T02:00:00.000Z',
        stops: ['FRA', 'BKK', 'SIN'],
        duration: 800,
      },
      {
        origin: 'HKT',
        destination: 'MOW',
        date: '2018-05-18T08:30:00.000Z',
        stops: ['SIN', 'BKK', 'FRA'],
        duration: 750,
      },
    ],
  },
];
