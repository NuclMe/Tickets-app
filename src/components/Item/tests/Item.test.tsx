import { render } from '@testing-library/react';

import { Item } from '../Item';

import { mockData } from './mockData';

describe('Item', () => {
  it('renders the Item component via snapshot', () => {
    render(
      <Item key={1} price={14000} carrier="S7" segments={mockData.segments} />
    );

    expect(Item).toMatchSnapshot();
  });
});
