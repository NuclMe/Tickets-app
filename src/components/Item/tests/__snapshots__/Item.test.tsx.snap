// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Item > renders the Item component via snapshot 1`] = `[Function]`;
